flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -401,6 +403,22 @@
 			.withDescription("Fraction of Total Flink Memory to be used as Managed Memory, if Managed Memory size is not"
 				+ " explicitly specified.");
 
+	/**
+	 * Weights of managed memory consumers.
+	 */
+	// Do not advertise this option until the feature is completed.
+	@Documentation.ExcludeFromDocumentation
+	public static final ConfigOption<Map<String, String>> MANAGED_MEMORY_CONSUMER_WEIGHTS =
+		key("taskmanager.memory.managed.consumer-weights")
+			.mapType()
+			.defaultValue(new HashMap<String, String>() {{
+				put("DATAPROC", "70");
+				put("PYTHON", "30");
```

Comment:
It would be good to have an `enum` which holds all supported kinds.

Reply:
Since map type `ConfigOption` only accepts `Map<String, String>`, I'll use a static inner class with static final string constants for the consumer names, to avoid converting enum values to strings.